tools/wpt/run.py

```diff
@@ -775,15 +775,19 @@ def setup_wptrunner(venv, prompt=True, install_browser=False, **kwargs):
         else:
             logger.info("Valid channels for %s not known; using argument unmodified" % kwargs["product"])
             kwargs["browser_channel"] = kwargs["channel"]
-        del kwargs["channel"]
 
-    if install_browser:
+    if kwargs["install_browser"]:
         logger.info("Installing browser")
         kwargs["binary"] = setup_cls.install(channel=channel)
 
-    setup_cls.setup(kwargs)
+    # Remove kwargs we handle here
+    wptrunner_kwargs = kwargs.copy()
+    for kwarg in ["install_browser", "install_webdriver", "channel"]:
+        del kwargs[kwarg]
 
-    wptcommandline.check_args(kwargs)
+    setup_cls.setup(wptrunner_kwargs)
+
+    wptcommandline.check_args(wptrunner_kwargs)
```

Comment:
I thought this was meant to check only the kwargs known by wptrunner. In the existing code, all flags specific to `wpt run` have been removed from `kwargs` at this point, and this changes that.

Besides, this is still not consistent. `prompt` and `affected` are handled differently.

I'd suggest we either pop all four `wpt run` flags in `run` and explicitly pass them
`setup_wptrunner(venv, prompt, affected, install_browser, install_webdriver, **kwargs)`

or make `setup_wptrunner` take just `(venv, **kwargs)` and pop all four flags here.